test/plans/test_constrained_plan.jl

```diff
@@ -72,49 +91,49 @@ include("../utils/dummy_types.jl")
         )
         co1f = ConstrainedManifoldObjective(f, grad_f!; g=g, grad_g=grad_g)
```

Comment:
```suggestion
        co1f = ConstrainedManifoldObjective(f, grad_f!; g=g, grad_g=grad_g, M=M)
```
Automatic detection doesn't always work well when there is no manifold given.

Reply:
I have not worked through these, I think; but yes, with `p` being also just `rand(M)` if it is not nothing, this probably helps here. I was focussing a bit on counting and general stupid typos today, but now the rest of the day is meetings and forms.